breacharbiter.go

```diff
@@ -944,17 +944,24 @@ func newRetributionInfo(chanPoint *wire.OutPoint,
 
 	// First, record the breach information for the local channel point if
 	// it is not considered dust, which is signaled by a non-nil sign
-	// descriptor. Here we use CommitmentNoDelay since this output belongs
-	// to us and has no time-based constraints on spending.
+	// descriptor. Here we use CommitmentNoDelay (or
+	// CommitmentNoDelayTweakless for newer commitments) since this output
+	// belongs to us and has no time-based constraints on spending.
 	if breachInfo.LocalOutputSignDesc != nil {
+		witnessType := input.CommitmentNoDelay
+		if breachInfo.LocalOutputSignDesc.SingleTweak == nil {
```

Comment:
Seems a bit arbitrary to use the tweak value to signal tweakless channel here, but the channel type other places. Should make a decision which approach to go with and use that everywhere? (would possibly require chan type to be added to the breach info).

Reply:
This doesn't have access to the channel type nor the channel state. Doing so would require more changes in the chain watcher and also channel state machine to thread through all the state.